tests/lib/validator_test.py

```diff
@@ -290,6 +291,72 @@ def _create_route1(self):
         )
 
 
+class TestVxlanValidation(object):
+    def test_missing_parent_is_invalid(self):
```

Comment:
The scenarios look great.
These tests that check missing required parameters look to have the same template, so you can use pytest parametrize option and do something like this:
```
parametrize_vxlan_req_fields = pytest.mark.parametrize(
    'required_field', [VXLAN.FOO, VXLAN.FOO2, ...]
)

    @parametrize_vxlan_req_fields
    def test_missing_parent_is_invalid(self, required_field):
        vxlan_state = {
            VXLAN.ID: 99,
            VXLAN.REMOTE: '192.168.3.3',
            VXLAN.BASE_IFACE: "eth0",
        }
        vxlan_state.pop(required_fields)

        desired_state = {
            schema.Interface.KEY: [
                {
                    schema.Interface.NAME: 'eth0.101',
                    schema.Interface.TYPE: VXLAN.TYPE,
                    VXLAN.CONFIG_SUBTREE: vxlan_state,
                }
            ]
        }
        with pytest.raises(NmstateValueError):
            libnmstate.validator.validate_vxlan(desired_state)
```